### PR TITLE
[types/parse] User type updated to include requestEmailVerification function. 

### PR DESCRIPTION
**Pull request by rsmets**
Parse documentation here: https://parseplatform.org/Parse-SDK-JS/api/3.2.0/Parse.User.html\#.requestEmailVerification

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://parseplatform.org/Parse-SDK-JS/api/3.2.0/Parse.User.html\#.requestEmailVerification>

### CODE REVIEW

File: types/parse/index.d.ts [hunk 1]
```diff
@@ -1016,6 +1016,7 @@ declare global {
             logIn<T extends User>(username: string, password: string, options?: FullOptions): Promise<T>;
             logOut<T extends User>(): Promise<T>;
             requestPasswordReset<T extends User>(email: string, options?: SuccessFailureOptions): Promise<T>;
+            requestEmailVerification<T extends User>(email: string, options?: SuccessFailureOptions): Promise<T>;
```
Comment (RaschidJFR) [inline types/parse/index.d.ts:1019]:
```suggestion
            requestEmailVerification<T extends User>(email: string, options?: RequestOptions): Promise<T>;
```

Comment (rsmets) [reply types/parse/index.d.ts:1019]: Thank you @RaschidJFR. Suggestion committed.

**Comment by RaschidJFR** [comment]
No prob, thank you @rsmets for your contribution. I'm just unsure about what to do with other ts version indexes. We've been adding changes also in [/ts3.6](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/parse/ts3.6/index.d.ts) for backward compatibility, but at this point I'm unsure if we should keep doing it. 

I'll dig into previous PRs to see how it has been managed lately, I haven't been around for a while. It may take me some time, so if you don't mind, I think it'll be safer if you also add the same changes inside the ts.36 folder. Would you mind doing that?

**Comment by rsmets** [comment]
I was unsure what todo regarding the version indexes as well... glad to know I am not the only one!

Just updated the ts.36 directory to include the functionality as well. 👍🏼

**Review by RaschidJFR — APPROVED** [review]

**Comment by rsmets** [comment]
Ready to merge

**Comment by RaschidJFR** [comment]
Hey @rdhelms @dplewis when do you think we should drop support for ts3.6?

**Pull request merged**